jwst/assign_wcs/nirspec.py

```diff
@@ -1693,6 +1697,31 @@ def _create_ifupost_transform(ifupost_slice):
     model = linear & Identity(1) | model_poly
     return model
 
+def autowave(input_model, reference_files, slit_y_range):
+    """Return the appropriate function for autowave data
+
+    Parameters
+    ----------
+
+    input_model : `~jwst.datamodels.DataModel`
+        The input data model.
+
+    reference_files : dict
+        The reference files used for this mode.
+
+    slit_y_range : list
+        The slit dimensions relative to the center of the slit.
+
+    """
+    lamp_mode = input_model.meta.instrument.lamp_mode.lower()
+    if lamp_mode in ['fixedslit', 'brightobj']:
+        return slits_wcs(input_model, reference_files, slit_y_range)
+    elif lamp_mode == 'ifu':
+        return ifu(input_model, reference_files, slit_y_range)
+    elif lamp_mode == 'msaspec':
+        return slits_wcs(input_model, reference_files, slit_y_range)
+    else:
+        return not_implemented_mode
```

Comment:
Where is `not_implemented_mode` defined/set?

Reply:
The `not_implemented_mode` function below requires 2 arguments.

Reply:
Agreed it needs arguments. 3 of them actually

Reply:
IIUC, `not_implemented_mode` is not intended to actually be evaluated at the point where it is returned by `autowave`.  `autowave` normally returns a pipeline, but in the case of `not_implemented_mode`, `autowave` returns a function (that returns None if it is eventually evaluated).  So no, you don't want to add an argument list to it here.

Reply:
Yes but when `not_implemented_mode` returned by `autowave`, the arguments have been consumed in the `autowave` call, so they need to be put back when the function call is passed on